android/src/main/java/com/thunderhead/one_sdk_flutter/OnePlugin.java

```diff
@@ -29,132 +40,243 @@
+                    .build();
+
+            try {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                    One.sendResponseCode(responseCodeRequest);
+                    result.success(null);
+                } else {
+                    One.sendResponseCodeLegacySupport(responseCodeRequest);
+                    result.success(null);
+                }
+            } catch (ExecutionException error) {
+                result.error(Integer.toString(error.hashCode()), error.getLocalizedMessage(), error);
+                error.printStackTrace();
+            }
+        });
+    }
+
+    private void optOut(MethodCall call, final Result result) {
+        OneOptOutConfiguration optOutConfiguration;
+
+        final HashMap<String, Object> arguments = call.arguments();
+        Boolean optOut = (Boolean) arguments.get("optOut");
+        List<String> options = (List<String>) arguments.get("options");
+
+        if (optOut == null) {
+            optOut = false;
+        }
+
+        if (options != null && !options.isEmpty()) {
+            for (String optOutOption : options) {
+                if (optOutOption.equals("cityCountryDetection")) {
+                    Set<OneOptInOptions> optInOptions = EnumSet.noneOf(OneOptInOptions.class);
+                    optInOptions.add(OneOptInOptions.CITY_COUNTRY_DETECTION);
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .optInOptions(optInOptions)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
+                }
+                if (optOutOption.equals("allTracking")) {
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
+                }
+            }
+        } else {
+            optOutConfiguration = new OneOptOutConfiguration.Builder()
+                    .optOut(optOut)
+                    .build();
+            One.setOptOutConfiguration(optOutConfiguration);
+        }
+        result.success(null);
+    }
+
+    // Helper methods
+
+    // Convert OneResponse Class to HashMap so Dart can read it.
+    private HashMap<String, Object> responseObjectToHashMap(OneResponse response) {
+        HashMap<String, Object> responseMap = new HashMap<>();
+        responseMap.put("tid", response.getTid());
+        responseMap.put("interactionPath", response.getInteractionPath().getValue().getPath());
+
+        if (!response.getOptimizationPoints().isEmpty()) {
+            List<HashMap<String, Object>> optimizationsList = new ArrayList<>();
+            HashMap<String, Object> optimizationPointMap = new HashMap<>();
+
+            for (OptimizationPoint point : response.getOptimizationPoints()) {
```

Comment:
this is wrong.  You need a new optimizationPointMap each time otherwise you will end up with a list of points all with the same data as each item in the optimizationsList points to the same optimizationPointMap.

See this example of what your code is doing: https://pl.kotl.in/cwjRmKTdl?readOnly=true